x-pack/packages/kbn-elastic-assistant/impl/assistant/use_conversation/index.tsx

```diff
@@ -52,298 +49,265 @@ interface CreateConversationProps {
+      if (isHttpFetchError(currentConversation)) {
+        return;
+      }
+      return currentConversation;
+    },
+    [http]
+  );
+
   /**
    * Removes the last message of conversation[] for a given conversationId
    */
   const removeLastMessage = useCallback(
-    (conversationId: string) => {
+    async (conversationId: string) => {
       let messages: Message[] = [];
-      setConversations((prev: Record<string, Conversation>) => {
-        const prevConversation: Conversation | undefined = prev[conversationId];
-
-        if (prevConversation != null) {
-          messages = prevConversation.messages.slice(0, prevConversation.messages.length - 1);
-          const newConversation = {
-            ...prevConversation,
-            messages,
-          };
-          return {
-            ...prev,
-            [conversationId]: newConversation,
-          };
-        } else {
-          return prev;
-        }
-      });
+      const prevConversation = await getConversationById({ http, id: conversationId });
+      if (isHttpFetchError(prevConversation)) {
+        return;
+      }
+      if (prevConversation != null) {
+        messages = prevConversation.messages.slice(0, prevConversation.messages.length - 1);
+        await updateConversationApi({
+          http,
+          conversationId,
+          messages,
+        });
+      }
       return messages;
     },
-    [setConversations]
+    [http]
   );
 
   /**
    * Updates the last message of conversation[] for a given conversationId with provided content
    */
   const amendMessage = useCallback(
-    ({ conversationId, content }: AmendMessageProps) => {
-      setConversations((prev: Record<string, Conversation>) => {
-        const prevConversation: Conversation | undefined = prev[conversationId];
-
-        if (prevConversation != null) {
-          const { messages, ...rest } = prevConversation;
-          const message = messages[messages.length - 1];
-          const updatedMessages = message
-            ? [...messages.slice(0, -1), { ...message, content }]
-            : [...messages];
-          const newConversation = {
-            ...rest,
-            messages: updatedMessages,
-          };
-          return {
-            ...prev,
-            [conversationId]: newConversation,
-          };
-        } else {
-          return prev;
-        }
-      });
+    async ({ conversationId, content }: AmendMessageProps) => {
+      const prevConversation = await getConversationById({ http, id: conversationId });
+      if (isHttpFetchError(prevConversation)) {
+        return;
+      }
+      if (prevConversation != null) {
+        const { messages } = prevConversation;
+        const message = messages[messages.length - 1];
+        const updatedMessages = message ? [{ ...message, content }] : [];
+
+        await appendConversationMessagesApi({
+          http,
+          conversationId,
+          messages: updatedMessages,
+        });
```

Comment:
The return value of `appendConversationMessagesApi` is

```ts
Promise<Conversation | IHttpFetchError>
```

but it's ambiguous whether or not it's ok to not test for a return value of type `IHttpFetchError` in this case, as detailed in https://github.com/elastic/kibana/pull/173487/files#r1465643677 , because the error won't propagate up the stack

Reply:
Resolved by commit https://github.com/elastic/kibana/pull/173487/commits/2ea2fb7a89d35ea7e3e920216ecb20b65210f1a3